packages/xcm-analyser/src/types.ts

```diff
@@ -0,0 +1,127 @@
+import { z } from 'zod';
+
+export type JunctionType =
+  | 'Parachain'
+  | 'AccountId32'
+  | 'AccountIndex64'
+  | 'AccountKey20'
+  | 'PalletInstance'
+  | 'GeneralIndex'
+  | 'GeneralKey'
+  | 'OnlyChild'
+  | 'Plurality'
+  | 'GlobalConsensus';
+
+const JunctionParachain = z.object({ Parachain: z.string() });
+const JunctionAccountId32 = z.object({
+  AccountId32: z.object({ network: z.string().nullable(), id: z.string() }),
+});
+const JunctionAccountIndex64 = z.object({
+  AccountIndex64: z.object({ network: z.string().nullable(), index: z.string() }),
+});
+const JunctionAccountKey20 = z.object({
+  AccountKey20: z.object({ network: z.string().nullable(), key: z.string() }),
+});
+const JunctionPalletInstance = z.object({ PalletInstance: z.string() });
```

Comment:
For all the number types lets use

`z.number().or(z.string()).pipe(z.coerce.number())`

Reply:
Actually we might want to use bigint since some of these numbers can be u128

> GeneralIndex = 5 (Compact<u128>): A nondescript index within the context location. Usage will vary widely owing to its generality. Note: Try to avoid using this and instead use a more specific item.

Reply:
Hello @ryanleecode , I made changes to schema to accept string, numbers and bigints using `z.string().or(z.number()).or(z.bigint())`. I tried using coerce but then the inferred TS types required bigints everwhere. Can you check out the changes?

Now the string numbers values can be entered like string or number:

```
const multilocation: MultiLocation = {
  parents: '1',
  interior: {
    X2: [{ PalletInstance: '50' }, { GeneralIndex: 1984 }],
  },
};
```

Or BigInt type:

```
const multilocation: MultiLocation = {
  parents: '1',
  interior: {
    X2: [{ PalletInstance: '50' }, { GeneralIndex: BigInt(1984) }],
  },
};
```

Reply:
Yeah this is fine. You can also add a regexp constraint `const StringOrNumber = z.string().regex(/^\d+$/).or(z.number()).or(z.bigint());`.

For `AccountId32.id`, `AccountKey20.key`, `GeneralKey.data`, change it back to string since they are hex encoded uint8arrays.

Reply:
Hey @ryanleecode, I have update the string or number schema to include regex and changed the `AccountId32.id`, `AccountKey20.key`, `GeneralKey.data` back to string.

Let us know if that's all.

Reply:
Perfect! Lets get this merged